Update schema descriptions, examples and regenerate the docs (#74)
EnterpriseDB/terraform-provider-biganimal: docs/resources/cluster.md

# biganimal_cluster (Resource)

> The cluster resource is used to manage BigAnimal clusters. See https://www.enterprisedb.com/docs/biganimal/latest/getting_started/creating_a_cluster/ for more details.

Author: The clickable links are possible, and they look like this in the terraform registry documentation. 
I'm generating clickable links for the rest of the links in a separate commit. 

![Screenshot 2022-11-28 at 17 04 07](https://user-images.githubusercontent.com/121254/204324824-239c3739-2459-43a7-85f1-4c76ca1fd357.png)

> The cluster resource is used to manage BigAnimal clusters. See https://www.enterprisedb.com/docs/biganimal/latest/getting_started/creating_a_cluster/ for more details.

Author: @dwicinas should we explicitly state that the `Creating a cluster` documentation is an `external` link to the terraform registry? Or should we explicitly state that it's an EDB BigAnimal documentation link?

  description = "The name of the cluster."
}

resource "biganimal_cluster" "eha_cluster" {
  cluster_name = var.cluster_name

  allowed_ip_ranges {
    cidr_block  = "127.0.0.1/32"
    description = "localhost"
  }

  allowed_ip_ranges {
    cidr_block  = "192.168.0.1/32"
    description = "description!"
  }

  backup_retention_period = "6d"
  cluster_architecture {
    id    = "eha"
    nodes = 5
  }

  instance_type = "aws:c5.large"
  password      = resource.random_password.password.result
  pg_config {
    name  = "application_name"
    value = "created through terraform"
  }

  pg_config {
    name  = "array_nulls"
    value = "off"
  }

  storage {
    volume_type       = "gp3"
    volume_properties = "gp3"
    size              = "4 Gi"
  }

  pg_type            = "epas"
  pg_version         = "14"
  private_networking = false
  cloud_provider     = "aws"
  region             = "us-east-1"
}

output "password" {
  sensitive = true
  value     = resource.biganimal_cluster.eha_cluster.password
}

output "connection_uri" {
  value = resource.biganimal_cluster.eha_cluster.connection_uri
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Required

> - `cloud_provider` (String) Cloud Provider. e.g. "aws" or "azure".

Reviewer: ```suggestion
- `cloud_provider` (String) Cloud Provider. For example, "aws" or "azure".
```

Author: Fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/4de4c956dd0831008781a5f59b9011542e378c5a

> - `cluster_architecture` (Block List, Min: 1) Cluster Architecture. See https://www.enterprisedb.com/docs/biganimal/latest/overview/02_high_availability/ for details. (see [below for nested schema](#nestedblock--cluster_architecture))

Reviewer: ```suggestion
- `cluster_architecture` (Block List, Min: 1) Cluster Architecture. See [Supported cluster types ](https://www.enterprisedb.com/docs/biganimal/latest/overview/02_high_availability/) for details. See also [schema information](#nestedblock--cluster_architecture).
```

Author: I will check the markdown clickable link part (probably it's doable). The `see [below for nested schema]` is coming from the provider generator, which I explained [in this comment](https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/74#discussion_r1033576476).

Author: Clickable link is added at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/0aea7392df32d1d3b7fcb6e35524aa585fa8a660

- `cluster_name` (String) Name of the cluster.

> - `instance_type` (String) Instance Type. e.g. "azure:Standard_D2s_v3", "aws:c5.large".

Reviewer: ```suggestion
- `instance_type` (String) Instance Type. For example, "azure:Standard_D2s_v3", "aws:c5.large".
```

Author: Fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/4de4c956dd0831008781a5f59b9011542e378c5a

- `password` (String, Sensitive) Password for the user edb_admin. It must be 12 characters or more.

> - `pg_type` (String) Postgres type. e.g. "epas", "pgextended" or "postgres".

Reviewer: ```suggestion
- `pg_type` (String) Postgres type. For example, "epas", "pgextended", or "postgres".
```

Author: `e.g.` -> `For example,` fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/4de4c956dd0831008781a5f59b9011542e378c5a
Serial comma, fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/a7e9930ede32c6adc1fc0810c56a60d8391f3fb3

> - `pg_version` (String) Postgres version. See https://www.enterprisedb.com/docs/biganimal/latest/overview/05_database_version_policy/#supported-postgres-types-and-versions for supported Postgres types and versions.

Reviewer: ```suggestion
- `pg_version` (String) Postgres version. See [Database version policy ](https://www.enterprisedb.com/docs/biganimal/latest/overview/05_database_version_policy/#supported-postgres-types-and-versions) for supported Postgres types and versions.
```

Author: Clickable link is added at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/0aea7392df32d1d3b7fcb6e35524aa585fa8a660

Also, related comment about this change: https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/74#discussion_r1033752771

> - `region` (String) Region to deploy the cluster. See https://www.enterprisedb.com/docs/biganimal/latest/overview/03a_region_support/ for supported regions.

Reviewer: ```suggestion
- `region` (String) Region to deploy the cluster. See [Supported regions ](https://www.enterprisedb.com/docs/biganimal/latest/overview/03a_region_support/) for supported regions.
```

Author: Clickable link is added at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/0aea7392df32d1d3b7fcb6e35524aa585fa8a660

- `storage` (Block List, Min: 1) Storage. (see [below for nested schema](#nestedblock--storage))

### Optional

- `allowed_ip_ranges` (Block List) Allowed IP ranges. (see [below for nested schema](#nestedblock--allowed_ip_ranges))

> - `backup_retention_period` (String) Backup Retention Period. e.g. "7d", "2w" or "3m".

Reviewer: ```suggestion
- `backup_retention_period` (String) Backup Retention Period. For example, "7d", "2w", or "3m".
```

Author: e.g. -> For example, fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/4de4c956dd0831008781a5f59b9011542e378c5a
Serial comma, fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/a7e9930ede32c6adc1fc0810c56a60d8391f3fb3

> - `pg_config` (Block List) Database Configuration Parameters. See https://www.enterprisedb.com/docs/biganimal/latest/using_cluster/03_modifying_your_cluster/05_db_configuration_parameters/ for details. (see [below for nested schema](#nestedblock--pg_config))

Reviewer: ```suggestion
- `pg_config` (Block List) Database Configuration Parameters. See [Modifying database configuration parameters ](https://www.enterprisedb.com/docs/biganimal/latest/using_cluster/03_modifying_your_cluster/05_db_configuration_parameters/) for details. For schema details, see [nested schema](#nestedblock--pg_config).
```

Author: Clickable link is added at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/0aea7392df32d1d3b7fcb6e35524aa585fa8a660

- `private_networking` (Boolean) Is private networking enabled.
- `read_only_connections` (Boolean) Is read only connection enabled.
- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))

### Read-Only

- `cluster_id` (String) Cluster ID.
- `connection_uri` (String) Cluster connection URI.
- `created_at` (String) Cluster Creation Time.
- `deleted_at` (String) Cluster Deletion Time.
- `expired_at` (String) Cluster Expiry Time.
- `first_recoverability_point_at` (String) Earliest Backup Recover Time.
- `id` (String) The ID of this resource.
- `phase` (String) Current Phase of the cluster.
- `resizing_pvc` (List of String) Resizing PVC.

> - `ro_connection_uri` (String) Cluster read-only connection URI, only available for high availability clusters.

Reviewer: ```suggestion
- `ro_connection_uri` (String) Cluster read-only connection URI. Only available for high availability clusters.
```

Author: Fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/289427cc4f5ea4aaa12f469e36fe7df21abe239d

<a id="nestedblock--cluster_architecture"></a>
### Nested Schema for `cluster_architecture`

Required:

> - `id` (String) Cluster Architecture ID. e.g. "single", "ha" or "eha".

Reviewer: ```suggestion
- `id` (String) Cluster Architecture ID. For example, "single", "ha", or "eha".
```

Author: e.g. -> For example, fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/4de4c956dd0831008781a5f59b9011542e378c5a
Serial comma, fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/a7e9930ede32c6adc1fc0810c56a60d8391f3fb3

- `nodes` (Number) Node Count.

Read-Only:

- `name` (String) Name.

<a id="nestedblock--storage"></a>
### Nested Schema for `storage`

Required:

- `size` (String) Size of the volume. It can be set to different values depending on your volume type and properties.
- `volume_properties` (String) Volume Properties in accordance to the selected volume type.

> - `volume_type` (String) Volume Type. For Azure: "azurepremiumstorage" or "ultradisk", for AWS: "gp3", "io2" or "io2-block-express".

Reviewer: ```suggestion
- `volume_type` (String) Volume Type. For Azure: "azurepremiumstorage" or "ultradisk". For AWS: "gp3", "io2", or "io2-block-express".
```
